kura/org.eclipse.kura.net2.admin/src/main/java/org/eclipse/kura/net2/admin/NMDbusConnector.java

```diff
@@ -0,0 +1,223 @@
+        return dbusConnection.getRemoteObject(NM_BUS_NAME, connectionPath.getPath(), Connection.class);
+    }
+
+    private Map<String, Variant<?>> copyConnectionSettings(Map<String, Map<String, Variant<?>>> connectionSettings) {
+        Map<String, Variant<?>> connectionMap = new HashMap<>();
+        for (String key : connectionSettings.get("connection").keySet()) {
+            connectionMap.put(key, connectionSettings.get("connection").get(key));
+        }
+
+        return connectionMap;
+    }
+
+    private Map<String, Variant<?>> buildIpv4Settings(Map<String, Object> networkConfiguration, String iface) {
+        Map<String, Variant<?>> ipv4Map = new HashMap<>();
+
+        String dhcpClient4EnabledProperty = String.format("net.interface.%s.config.dhcpClient4.enabled", iface);
+        Boolean dhcpClient4Enabled = (Boolean) networkConfiguration.get(dhcpClient4EnabledProperty);
+
+        // Should handle net.interface.eth0.config.ip4.status here
+
+        if (Boolean.FALSE.equals(dhcpClient4Enabled)) {
+            ipv4Map.put("method", new Variant<>("manual"));
+
+            String dhcpClient4AddressProperty = String.format("net.interface.%s.config.ip4.address", iface);
+            String dhcpClient4Address = (String) networkConfiguration.get(dhcpClient4AddressProperty);
+
+            String dhcpClient4PrefixProperty = String.format("net.interface.%s.config.ip4.prefix", iface);
+            Short dhcpClient4Prefix = (Short) networkConfiguration.get(dhcpClient4PrefixProperty);
+
+            Map<String, Variant<?>> address = new HashMap<>();
+            address.put("address", new Variant<>(dhcpClient4Address));
+            address.put("prefix", new Variant<>(new UInt32(dhcpClient4Prefix)));
+
+            List<Map<String, Variant<?>>> addressData = Arrays.asList(address);
+            ipv4Map.put("address-data", new Variant<>(addressData, "aa{sv}"));
+
+            String dhcpClient4DNSProperty = String.format("net.interface.%s.config.ip4.dnsServers", iface);
+            if (networkConfiguration.containsKey(dhcpClient4DNSProperty)) {
+                String dhcpClient4DNS = (String) networkConfiguration.get(dhcpClient4DNSProperty);
+                ipv4Map.put("ignore-auto-dns", new Variant<>(true));
+                ipv4Map.put("dns-search", new Variant<>(getDNSServers(dhcpClient4DNS)));
+            } else {
+                ipv4Map.put("ignore-auto-dns", new Variant<>(false));
+            }
+
+            String dhcpClient4GatewayProperty = String.format("net.interface.%s.config.ip4.gateway", iface);
+            if (networkConfiguration.containsKey(dhcpClient4GatewayProperty)) {
+                String dhcpClient4Gateway = (String) networkConfiguration.get(dhcpClient4GatewayProperty);
+                ipv4Map.put("gateway", new Variant<>(dhcpClient4Gateway));
+            }
+
+        } else {
+            ipv4Map.put("method", new Variant<>("auto"));
+        }
```

Comment:
We'd manage the situation when the interface is set as dhcp, but the user has set some custom dns servers.

Reply:
Yeah... this can be done with a combination of `method` set to `auto` and `ignore-auto-dns` set to false. I was planning to fix it later but I might do this now since I'm already at it xD